src/plugins/intel_cpu/src/emitters/snippets/x64/kernel_executors/brgemm_copy_b.cpp

```diff
@@ -0,0 +1,315 @@
+    brgCopyKernelConf.src_zp_type = dnnl::impl::cpu::x64::none;
+
+    std::unique_ptr<matmul::jit_brgemm_matmul_copy_b_t> kernel = nullptr;
+    OV_CPU_JIT_EMITTER_ASSERT(matmul::create_brgemm_matmul_copy_b(kernel, &brgCopyKernelConf) == dnnl_success,
+                              "cannot create kernel due to invalid params");
+    return kernel;
+}
+
+void BrgemmCopyBKernel::emit_brgemm_copy_b_kernel_call(size_t N, size_t K, size_t offset_in, size_t offset_out, size_t offset_comp) {
+    JitSafeInternalCall safe_internal_caller(this);
+
+    const auto add_offset = [&](Xbyak::Reg64 reg, size_t bytes_offset) {
+        if (bytes_offset) add(reg, bytes_offset);
+    };
+
+    // save function address in gpr to pass in call instruction
+    const auto& kernel_overload = static_cast<void (*)(matmul::jit_brgemm_matmul_copy_b_t*, const void*, const void*, const void*, size_t, size_t)>(execute);
+    mov(rbp, reinterpret_cast<uintptr_t>(kernel_overload));
+    mov(abi_param1, reinterpret_cast<uintptr_t>(kernel.get()));
+
+    add_offset(src_reg, offset_in); // abi_param2
+    add_offset(tr_src_reg, offset_out); // abi_param3
+    if (conf.is_with_comp())  // abi_param4
+        add_offset(comp_reg, offset_comp);
+    else
+        mov(comp_reg, reinterpret_cast<uintptr_t>(nullptr));
+
+#ifdef _WIN32
+    // Note: ABI requires that the remaining parameters (except the first for) are pushed to the stack in right-to-left order
+    //  Shadow space will be allocated inside internal_call_rsp_align()
+    push(K);
+    push(N);
+#else
+    mov(abi_param5, N);
+    mov(abi_param6, K);
+#endif
+
+    safe_internal_caller.call(rbp);
+
+#ifdef _WIN32
+    static constexpr int gpr_size = 8;
+    add(rsp, gpr_size * 2);
+#endif
+}
+
+void BrgemmCopyBKernel::execute(matmul::jit_brgemm_matmul_copy_b_t* kernel, const void* src, const void* dst, const void* comp, size_t N, size_t K) {
+    auto ctx = matmul::jit_brgemm_matmul_copy_b_t::ctx_t();
+    ctx.current_N_blk = N;
+    ctx.src = src;
+    ctx.tr_src = dst;
+    ctx.compensation_ptr = comp;
+    ctx.zp_a_compensation_ptr = nullptr;
+    ctx.zp_a_neg_value_ptr = nullptr;
+    ctx.current_K_start = 0;
+    ctx.current_K_iters = K;
+
+    OV_CPU_JIT_EMITTER_ASSERT(kernel, "Kernel hasn't been created");
+    (*kernel)(&ctx);
+}
+
+BrgemmCopyBKernelExecutor::BrgemmCopyBKernelExecutor(ov::intel_cpu::MultiCacheWeakPtr kernel_cache, BrgemmCopyBKernelConfig config)
+    : CPUKernelExecutor<BrgemmCopyBKernelConfig, BrgemmCopyBKernel>(std::move(kernel_cache), std::move(config)) { }
+
+std::shared_ptr<BrgemmCopyBKernel> BrgemmCopyBKernelExecutor::compile_kernel(const BrgemmCopyBKernelConfig& config) const {
+    std::shared_ptr<BrgemmCopyBKernel> compiled_kernel = std::make_shared<BrgemmCopyBKernel>();
+    // BrgemmCopyB is not executable - nothing to compile
+    if (!config.is_empty()) {
+        compiled_kernel = std::unique_ptr<BrgemmCopyBKernel>(new BrgemmCopyBKernel(config));
```

Comment:
Why do we initialize `shared_ptr` with `unique_ptr`? It seems like this appeared because of copy-paste from [brgemm kernel](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/emitters/snippets/x64/kernel_executors/brgemm.cpp#L158): but there we initialize `compiled_kernel`'s field with `unique_ptr`, not it itself

Also, I got confused while figuring out what is what among `compiled_kernel`, `kernel` and `ker_`. Could you please add a description for kernel related fileds of `BrgemmCopyBKernel`?

Reply:
> Why do we initialize shared_ptr with unique_ptr? It seems like this appeared because of copy-paste from [brgemm kernel](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/emitters/snippets/x64/kernel_executors/brgemm.cpp#L158): but there we initialize compiled_kernel's field with unique_ptr, not it itself

Agree that it's confused. I rewrote this part a little bit. Now we work only with `unique_ptr` as with all other emitters/kernels.

>Also, I got confused while figuring out what is what among compiled_kernel, kernel and ker_. Could you please add a description for kernel related fileds of BrgemmCopyBKernel?

Added brief comments and removed `conf` from the field of the `BrgemmCopyBKernel` to reduce memory size of this structure. Thanks!